CRM/Uimods/Tools/BankAccount.php

```diff
@@ -46,6 +51,44 @@ public static function renderForm($formName, &$form) {
         $form->assign('viewCustomData', $viewCustomData);
       }
     }
+
+    if ($formName == 'CRM_Custom_Form_CustomDataByType') {
+      if ($form->getVar('_type') == 'Contribution' && $form->getAction() == CRM_Core_Action::UPDATE) {
+        $to_ba_name = $from_ba_name = '';
+        foreach ($form->getVar('_groupTree') as $group) {
+          if ($group['name'] != 'contribution_information') {
+            continue;
+          }
+          foreach ($group['fields'] as $field) {
+            if ($field['name'] == 'to_ba') {
+              $to_ba_name = $field['element_name'];
+            }
+            if ($field['name'] == 'from_ba') {
+              $from_ba_name = $field['element_name'];
+            }
+          }
+        }
+
+        $contact_id = civicrm_api3('Contribution', 'getvalue', [
+          'return' => 'contact_id',
+          'id' => $form->getVar('_entityId'),
+        ]);
+
+        if ($form->elementExists($to_ba_name)) {
+          $label = $form->getElement($to_ba_name)->getLabel();
+          $form->removeElement($to_ba_name);
+          $options = ['' => ts('-- please select --')] + self::getBankAccountReferenceOptions($contact_id)['all_domain_ibans'];
+          $form->add('select', $to_ba_name, $label, $options, FALSE, ['class' => 'crm-select2']);
```

Comment:
You can use `['class' => 'crm-select2', 'placeholder' => ts('- none -'),]` instead of the empty element in `$option`.